test/gc.js

```diff
@@ -0,0 +1,55 @@
+'use strict'
+
+/* global WeakRef, FinalizationRegistry */
+
+const { test } = require('tap')
+const { Client } = require('..')
+const { createServer } = require('net')
+const FakeTimers = require('@sinonjs/fake-timers')
+
+const SKIP = typeof WeakRef === 'undefined' || typeof FinalizationRegistry === 'undefined'
+
+test('gc should collect the client if there are no active sockets', { skip: SKIP }, t => {
+  t.plan(3)
+
+  const clock = FakeTimers.install()
+  t.teardown(clock.uninstall.bind(clock))
+
+  const server = createServer((socket) => {
+    socket.write('HTTP/1.1 200 OK\r\n')
+    socket.write('Content-Length: 0\r\n')
+    socket.write('Keep-Alive: timeout=1s\r\n')
+    socket.write('Connection: keep-alive\r\n')
+    socket.write('\r\n\r\n')
+  })
+  t.teardown(server.close.bind(server))
+
+  const registry = new FinalizationRegistry(({ wrapper }) => {
+    wrapper.destroy().then(() => t.pass()).catch(t.error)
+  })
+
+  function wekRefTest (client, t) {
+    const weakRef = new WeakRef(client)
+    const wrapper = weakRef.deref()
+    wrapper.once('disconnect', () => {
+      t.pass()
+    })
+    registry.register(client, { wrapper })
+    wrapper.request({
+      path: '/',
+      method: 'GET'
+    }, (err, { body }) => {
+      t.error(err)
+      body.once('end', () => {
+        clock.tick(600)
+      }).resume()
+    })
+  }
```

Comment:
Maybe something like:

```js

t.plan(3)

const weakRef = new WeakRef(client)
let disconnected = false

const registry = new FinalizationRegistry((name) => {
  t.strictEqual(disconnected, true)
  t.strictEqual(name, 'test')
  t.strictEqual(weakRef.deref(), undefined)
})

client.once('disconnect', () => {
  disconnected = true
})

registry.register(client, 'test')

wrapper.request({
  path: '/',
  method: 'GET'
}, (err, { body }) => {
  t.error(err)
  body.once('end', () => {
    clock.tick(600)
  }).resume()
})
```

Reply:
It looks like it could be the fake clock

Reply:
Manually spinning the fake clock seems to trigger a GC operation. The test is taking more than 10 seconds, though.